### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by IvanGuculyak**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,68 @@
-# write your code here
+from __future__ import annotations
+
+
+class SoftwareEngineer:
+    def __init__(
+            self,
+            name: str
+    ) -> None:
```
Comment (katevasilenko) [inline app/main.py:8]: Please put those lines to one as it won’t be longer than flake8 allows you

File: app/main.py [hunk 2]
```diff
@@ -1 +1,68 @@
...
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(
+            self,
+            name: str
+    ) -> None:
```
Comment (katevasilenko) [inline app/main.py:21]: Please put those lines to one as it won’t be longer than flake8 allows you

File: app/main.py [hunk 3]
```diff
@@ -1 +1,68 @@
...
+    def create_awesome_web_page(self) -> str:
+        print(f"{self.name} is creating a webpage...")
+        return "<h1>Hello world</h1>"
+
+
+class BackendDeveloper(SoftwareEngineer):
+    def __init__(
+            self,
+            name: str,
+    ) -> None:
```
Comment (katevasilenko) [inline app/main.py:35]: Please put those lines to one as it won’t be longer than flake8 allows you

File: app/main.py [hunk 4]
```diff
@@ -1 +1,68 @@
...
+    def create_powerful_api(self) -> str:
+        print(f"{self.name} is creating an API...")
+        return "http://127.0.0.1:8000"
+
+
+class AndroidDeveloper(SoftwareEngineer):
+    def __init__(
+            self,
+            name: str,
+    ) -> None:
```
Comment (katevasilenko) [inline app/main.py:48]: Please put those lines to one as it won’t be longer than flake8 allows you

File: app/main.py [hunk 5]
```diff
@@ -1 +1,68 @@
...
+        super().__init__(name)
+        self.skills += ["Java", "Android studio"]
+
+    def create_smooth_mobile_app(self) -> str:
+        print(f"{self.name} is creating a mobile app...")
+        return "Ads every three swipes"
+
+
+class FullStackDeveloper(FrontendDeveloper, BackendDeveloper):
+    def __init__(
```
Comment (katevasilenko) [inline app/main.py:58]: You don’t need init were at all as inheritance do it automatically

File: app/main.py [hunk 6]
```diff
@@ -1 +1,68 @@
...
+            self,
+            name: str
+    ) -> None:
+        super().__init__(name)
+        self.skills = ["Python", "SQL", "Django", "JavaScript", "HTML", "CSS"]
+
+    def create_web_application(self) -> None:
+        print(f"{self.name} started creating a web application...")
+        BackendDeveloper.create_powerful_api(self)
+        FrontendDeveloper.create_awesome_web_page(self)
```
Comment (katevasilenko) [inline app/main.py:68]: Use `self` here

**Review by katevasilenko — CHANGES_REQUESTED** [review]

**Review by ArsenPidhoretskyi — APPROVED** [review]
Well done!

**Pull request closed without merging**